pkg/es/config/config.go

```diff
@@ -70,34 +81,56 @@ func (p IndexPrefix) Apply(indexName string) string {
 type Configuration struct {
-	Servers                  []string               `mapstructure:"server_urls" valid:"required,url"`
-	RemoteReadClusters       []string               `mapstructure:"remote_read_clusters"`
-	Username                 string                 `mapstructure:"username"`
-	Password                 string                 `mapstructure:"password" json:"-"`
-	TokenFilePath            string                 `mapstructure:"token_file"`
-	PasswordFilePath         string                 `mapstructure:"password_file"`
-	AllowTokenFromContext    bool                   `mapstructure:"-"`
-	Sniffer                  bool                   `mapstructure:"sniffer"` // https://github.com/olivere/elastic/wiki/Sniffing
-	SnifferTLSEnabled        bool                   `mapstructure:"sniffer_tls_enabled"`
-	MaxDocCount              int                    `mapstructure:"-"` // Defines maximum number of results to fetch from storage per query
-	MaxSpanAge               time.Duration          `mapstructure:"-"` // configures the maximum lookback on span reads
-	Timeout                  time.Duration          `mapstructure:"-"`
-	BulkSize                 int                    `mapstructure:"-"`
-	BulkWorkers              int                    `mapstructure:"-"`
-	BulkActions              int                    `mapstructure:"-"`
-	BulkFlushInterval        time.Duration          `mapstructure:"-"`
-	Indices                  Indices                `mapstructure:"indices"`
-	ServiceCacheTTL          time.Duration          `mapstructure:"service_cache_ttl"`
-	AdaptiveSamplingLookback time.Duration          `mapstructure:"-"`
-	Tags                     TagsAsFields           `mapstructure:"tags_as_fields"`
-	Enabled                  bool                   `mapstructure:"-"`
-	TLS                      configtls.ClientConfig `mapstructure:"tls"`
-	UseReadWriteAliases      bool                   `mapstructure:"use_aliases"`
-	CreateIndexTemplates     bool                   `mapstructure:"create_mappings"`
-	UseILM                   bool                   `mapstructure:"use_ilm"`
-	Version                  uint                   `mapstructure:"version"`
-	LogLevel                 string                 `mapstructure:"log_level"`
-	SendGetBodyAs            string                 `mapstructure:"send_get_body_as"`
+	// Servers is a list of Elasticsearch servers. The strings must must contain full URLs
+	// (i.e. http://localhost:9200).
+	Servers []string `mapstructure:"server_urls" valid:"required,url"`
+	// RemoteReadClusters is a list of Elasticsearch remote cluster names for cross-cluster
+	// querying.
+	RemoteReadClusters []string       `mapstructure:"remote_read_clusters"`
+	Authentication     Authentication `mapstructure:"auth"`
+	// TokenFilePath contains the path to a file containing a bearer token.
+	// This flag also loads CA if it is specified.
+	TokenFilePath string `mapstructure:"token_file"`
```

Comment:
Would you say token is also a form of authentication?

Reply:
@yurishkuro It looks like token is used for setting up the HTTP transport rather than authenticating the client itself (see https://github.com/jaegertracing/jaeger/blob/main/pkg/es/config/config.go#L482) which is why I left it out of the authentication config. Let me know if you feel like it belongs there though.

Reply:
my point was that bearer token is a specific form of authentication. I can see it being inside auth:
```yaml
auth:
  bearer_token:
    file_path:
    from_context: true
```
The doc string for the type should also explain the relationship between these two fields - I assume they are not mutually exclusive but also not coupled. What happens if both file is present and context has a token?